lib/promoted/ruby/client.rb

```diff
@@ -133,6 +134,13 @@ def deliver args, headers={}
 
           only_log = delivery_request_builder.only_log != nil ? delivery_request_builder.only_log : @default_only_log
           deliver_err = false
+
+          if !@pager.validate_paging(delivery_request_builder.full_insertion, delivery_request_builder.request[:paging])
+            # Invalid input, log and do SDK-side delivery.
+            @logger.warn("Invalid paging parameters") if @logger
+            only_log = true
```

Comment:
This is nice.  I wasn't expecting this.  I figured we just wouldn't record it if there weren't any response insertions.  I'm okay with this.  Eventually, we can have separate log records for the sdk inputs and move this over there.

Reply:
Where do we check for the insertion_paging_type is unpaged?  It'd be good to check that before this call this.  If we don't do the check, if someone passes in a paged list, I think it'd flag the validation case here and might fail silently.

Reply:
I might be misreading.

Reply:
I am following up with a separate PR to make the page type check work like in TS, and to not log.